src/components/public-link/create.vue

```diff
@@ -0,0 +1,109 @@
+except according to the terms contained in the LICENSE file.
+-->
+<template>
+  <modal id="public-link-create" :state="state" :hideable="!awaitingResponse"
+    backdrop @hide="$emit('hide')" @shown="$refs.displayName.focus()">
+    <template #title>{{ $t('title') }}</template>
+    <template #body>
+      <p class="modal-introduction">TODO</p>
+      <form @submit.prevent="submit">
+        <form-group ref="displayName" v-model.trim="displayName"
+          :placeholder="$t('field.displayName')" required autocomplete="off"/>
+        <div class="checkbox">
+          <label>
+            <input v-model="multiple" type="checkbox"
+              aria-describedby="public-link-create-multiple-help">
+              {{ $t('field.multiple') }}
+          </label>
+          <p id="public-link-create-multiple-help" class="help-block">
+            {{ $t('multipleHelp') }}
+          </p>
+        </div>
+        <div class="modal-actions">
+          <button type="submit" class="btn btn-primary"
+            :disabled="awaitingResponse">
+            {{ $t('action.create') }} <spinner :state="awaitingResponse"/>
+          </button>
+          <button type="button" class="btn btn-link"
+            :disabled="awaitingResponse" @click="$emit('hide')">
+            {{ $t('action.cancel') }}
+          </button>
+        </div>
+      </form>
+    </template>
+  </modal>
+</template>
+
+<script>
+import FormGroup from '../form-group.vue';
+import Modal from '../modal.vue';
+import Spinner from '../spinner.vue';
+import request from '../../mixins/request';
+import { apiPaths } from '../../util/request';
+import { noop } from '../../util/util';
+import { requestData } from '../../store/modules/request';
+
+export default {
+  name: 'PublicLinkCreate',
+  components: { FormGroup, Modal, Spinner },
+  mixins: [request()],
+  props: {
+    state: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data() {
+    return {
+      awaitingResponse: false,
+      displayName: '',
+      multiple: false
+    };
+  },
+  // The modal assumes that this data will exist when the modal is shown.
+  computed: requestData(['form']),
+  watch: {
+    state(state) {
+      if (!state) {
+        this.displayName = '';
+        this.multiple = false;
+      }
+    }
+  },
+  methods: {
+    submit() {
+      this.request({
+        method: 'POST',
+        url: apiPaths.publicLinks(this.form.projectId, this.form.xmlFormId),
+        data: { displayName: this.displayName, once: !this.multiple }
+      })
+        .then(({ data }) => {
+          this.$emit('success', data);
+        })
+        .catch(noop);
+    }
+  }
+};
+</script>
+
+<i18n lang="json5">
+{
+  "en": {
+    // This is the title at the top of a pop-up.
+    "title": "Create Public Access Link",
+    "field": {
+      "multiple": "Multiple responses"
+    },
+    "multipleHelp": "Allow this Form to be submitted more than once from the same device."
```

Comment:
Maybe this should make it a little clearer that this has to do with the link rather than with the Form? "Allow this link to be used to submit the Form more than once from the same device."? Or is that just too pedantic since the context makes it clear?

Reply:
I like having a shorter sentence there, but we could probably make the text clearer. What do you think of "Allow a User to submit more than once from the same device.", which doesn't mention the form? Or "Allow a User with this Link to submit more than once from the same device.", which mentions the link?

Reply:
Of course! "Allow a User with this Link to submit more than once from the same device." sounds great to me.